src/Opaleye/Internal/Sql.hs

```diff
@@ -108,6 +118,7 @@ baseTable ti columns = SelectFrom $
               , tables = [Table (HSql.SqlTable (PQ.tiSchemaName ti) (PQ.tiTableName ti))] }
 
 product :: NEL.NonEmpty Select -> [HPQ.PrimExpr] -> Select
+product (t NEL.:| []) [] = t
```

Comment:
The change probably comes from a larger branch I'm working on, which is a set of transformations on `From` (and friends) to simplify the generated queries but retain semantics. More on that in the future though.